test/integration/scheduler_perf/util.go

```diff
@@ -73,6 +74,37 @@ func newDefaultComponentConfig() (*config.KubeSchedulerConfiguration, error) {
 	return &cfg, nil
 }
 
+// clientSet is a combination of several different typed interfaces.
+// Because of name clashes this cannot be a simple struct with
+// three fields.
```

Comment:
@kerthcet: how do you like the idea of combining different clients in one interface?

It's not specific to scheduler_perf. A better place might be test/util/ktesting and (eventually) moving that package into staging so that we can use it in all of k/k.

Reply:
See https://github.com/kubernetes/kubernetes/pull/122481

Reply:
Let's wait for that PR before deciding about this one here.

Reply:
A combined interface for client looks good to me.

Reply:
This is neither here nor there. Let's do either `TContext` from https://github.com/kubernetes/kubernetes/pull/122481 or use completely separate parameters.

Reply:
I had to rebase and used the opportunity to update this PR with the "separate parameters" approach. I think using `TContext` is simpler *and* more powerful (for example, proper explanations for "context canceled").

I also updated https://github.com/kubernetes/kubernetes/pull/122481.